igvc_platform/src/tests/test_differential_drive.cpp

```diff
@@ -0,0 +1,202 @@
+//
+// Created by vivek on 3/11/20.
+//
+
+#include <gtest/gtest.h>
+#include <igvc_msgs/velocity_pair.h>
+#include <ros/ros.h>
+#include <geometry_msgs/Twist.h>
+#include <mocking_utils/mock_subscriber.h>
+
+class TestDifferentialDrive : public testing::Test
+{
+public:
+  TestDifferentialDrive() : mock_joy_pub(handle.advertise<geometry_msgs::Twist>("/cmd_vel", 1))
+  {
+  }
+
+protected:
+  [[nodiscard]] bool waitForSubscriber() const {
+    const double timeout = 5.0;
+    const double sleep_time = 1.0;
+    ros::Time end = ros::Time::now() + ros::Duration(timeout);
+    while (mock_joy_pub.getNumSubscribers() == 0)
+    {
+      ros::Duration(sleep_time).sleep();
+      if (ros::Time::now() > end)
+      {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  ros::NodeHandle handle;
+  ros::Publisher mock_joy_pub;
+};
+
+geometry_msgs::Twist createTwistMsg(float forward, float spin)
+{
+  geometry_msgs::Twist twist_msg;
+  twist_msg.angular.x = 0.0;
+  twist_msg.angular.y = 0.0;
+  twist_msg.angular.z = spin;
+  twist_msg.linear.x = forward;
+  twist_msg.linear.y = 0.0;
+  twist_msg.linear.z = 0.0;
+
```

Comment:
You should also test to make sure that the differential drive node ignores anything in the `angular.x`, `angular.y`, `linear.y` and `linear.z` fields

Reply:
Added a new test to make sure the differential drive node ignores the extraneous data.